reconcile/automated_actions/config/integration.py

```diff
@@ -0,0 +1,286 @@
+        self, query_func: Callable
+    ) -> Generator[AutomatedActionsInstanceV1, None, None]:
+        """Return all automated actions."""
+        data = instance_query(query_func, variables={})
+        for instance in data.automated_actions_instances_v1 or []:
+            if instance.deployment.delete:
+                continue
+            instance.permissions = list(
+                self._filter_permissions(instance.permissions or [])
+            )
+            yield instance
+
+    def _is_enabled(self, argument: AutomatedActionArgumentV1) -> bool:
+        """Check if the integration is enabled for the given argument namespace."""
+        if isinstance(argument, AutomatedActionArgumentOpenshiftV1):
+            return (
+                integration_is_enabled("automated-actions", argument.namespace.cluster)
+                and not argument.namespace.delete
+            )
+        return True
+
+    def _filter_permissions(
+        self, permissions: Iterable[PermissionAutomatedActionsV1]
+    ) -> Generator[PermissionAutomatedActionsV1, None, None]:
+        """Filter out expired roles and arguments (cluster.namespace) with disabled integrations."""
+        for permission in permissions:
+            # automated actions disabled for the cluster?
+            permission.arguments = [
+                arg for arg in permission.arguments or [] if self._is_enabled(arg)
+            ]
+            # remove expired roles
+            permission.roles = expiration.filter(permission.roles)
+            if not permission.roles:
+                continue
+            yield permission
+
+    def _compile_roles(
+        self, permissions: Iterable[PermissionAutomatedActionsV1]
+    ) -> list[AutomatedActionsRole]:
+        """Compile all automated actions roles."""
+        roles: list[AutomatedActionsRole] = []
+        for permission in permissions:
+            for role in permission.roles or []:
+                roles.extend(
+                    AutomatedActionsRole(user=user.org_username, role=role.name)
+                    for user in (role.users or []) + (role.bots or [])
+                )
+        return roles
+
+    def _compile_policies(
+        self, permissions: Iterable[PermissionAutomatedActionsV1]
+    ) -> list[AutomatedActionsPolicy]:
+        """Compile all automated actions policies."""
+        policies: list[AutomatedActionsPolicy] = []
+        for permission in permissions or []:
+            obj = permission.action.operation_id
+
+            parameters: list[dict[str, str]] = [] if permission.arguments else [{}]
+            for arg in permission.arguments or []:
+                match arg:
+                    case AutomatedActionArgumentOpenshiftV1():
+                        parameters.append({
+                            "cluster": arg.namespace.cluster.name,
+                            "namespace": arg.namespace.name,
+                            "kind": arg.kind_pattern or "Deployment|Pod",
```

Comment:
not a fan of this default :)

Reply:
Changed in [067950f](https://github.com/app-sre/qontract-reconcile/pull/4901/commits/067950fcb135c1a1eea13010cafb68bd69309288)